### PR TITLE
Remove actor dsl

### PR DESCRIPTION
**Pull request by chbatey**
https://github.com/akka/akka/issues/26186

### CODE REVIEW

**Comment by raboof** [comment]
Failures are Java compilation failures in `akka-docs/src/test/java/jdocs/actor/InboxDocTest.java`

File: akka-actor/src/main/mima-filters/2.5.x.backwards.excludes [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ProblemFilters.exclude[MissingClassProblem]("akka.actor.Inbox$")
```
Comment (raboof) [inline akka-actor/src/main/mima-filters/2.5.x.backwards.excludes]: Wow does this work? `2.5.x.backwards.excludes`?

File: akka-docs/src/test/scala/docs/io/EchoServer.scala [hunk 2]
```diff
@@ -10,32 +10,10 @@ import scala.concurrent.duration.DurationInt
 
 import com.typesafe.config.ConfigFactory
 
-import akka.actor.{ Actor, ActorDSL, ActorLogging, ActorRef, ActorSystem, Props, SupervisorStrategy }
-import akka.actor.ActorDSL.inbox
+import akka.actor.{ Actor, ActorLogging, ActorRef, ActorSystem, Props, SupervisorStrategy }
 import akka.io.{ IO, Tcp }
 import akka.util.ByteString
 
-object EchoServer extends App {
```
Comment (raboof) [inline akka-docs/src/test/scala/docs/io/EchoServer.scala]: Perhaps we should rewrite this one instead of removing?

Comment (chbatey) [reply akka-docs/src/test/scala/docs/io/EchoServer.scala]: :+1: wasn't included in the docs but useful to have a main users can run if they go to source

**Review by raboof — APPROVED** [review]

File: akka-actor/src/main/mima-filters/2.5.x.backwards.excludes (hunk 1 above)
Comment (chbatey) [reply akka-actor/src/main/mima-filters/2.5.x.backwards.excludes]: Yes but I think it should be 2.6.x instead of 2.5.x

Comment (chbatey) [reply akka-actor/src/main/mima-filters/2.5.x.backwards.excludes]: Or not!

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**